lib/escalate/mixin.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Escalate
+  module Mixin
+    class << self
+      def included(base)
+        raise 'instead of `include ExEscalator::Mixin`, you should `include ExEscalator.mixin`'
+      end
+    end
+
+    attr_accessor :escalate_logger_block
+
+    def ex_escalate(exception, message, **context)
+      Escalate.escalate(exception, message, escalate_logger, **context)
+    end
+
+    private
+
+    def escalate_logger
+      escalate_logger_block.try(:call) || default_escalate_logger
```

Comment:
Are there cases where we'll have a block but `call` won't work? If not, this could be simplified to
```
escalate_logger_block&.call || default_escalate_logger

Reply:
I put this in in the off case that executing the block itself raises an exception to make sure that we never hide an exception.

Reply:
But if it raises an exception, the rest of the line won't run. So I don't think that argues for `try`.

The only reason to use `try` is if you think a truthy object might be returned that doesn't respond to `call`. If that ever happened, I'd much rather that Ruby raise an exception telling us so. So I still think the `&.` above is the right choice.

Reply:
I keep forgetting that `try` doesn't catch exceptions, only catches if the method doesn't exist. I've moved to the `&.` approach you mentioned

009eeec338aba14c7ba5cb4073eafb4d88c4cbbf